webpack/InsightsHostDetailsTab/InsightsTotalRiskChart.js

```diff
@@ -0,0 +1,155 @@
+  const hashHistory = useHistory();
+  const dispatch = useDispatch();
+  const API_KEY = `HOST_${id}_RECOMMENDATIONS`;
+  const status = useSelector(state => selectAPIStatus(state, API_KEY));
+  const error = useSelector(state => selectAPIErrorMessage(state, API_KEY));
+  const { hits = [] } = useSelector(state => selectAPIResponse(state, API_KEY));
+
+  useEffect(() => {
+    id &&
+      dispatch(
+        get({
+          key: API_KEY,
+          url: insightsCloudUrl(`hits/${id}`),
+        })
+      );
+  }, [API_KEY, dispatch, id]);
+
+  // eslint-disable-next-line consistent-return
+  useEffect(() => {
+    if (!hits || !hits.length) return setTotalRisks(initialRisks);
+    const risks = { ...initialRisks };
+    // eslint-disable-next-line camelcase
+    hits.forEach(({ total_risk }) => {
+      risks[total_risk].value += 1;
+    });
+    risks.total = hits.length;
+    setTotalRisks(risks);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [hits.length]);
+
+  const onChartClick = (evt, { index }) => {
+    hashHistory.push(`/Insights`);
+    dispatch(
+      push({
+        search: `search=total_risk+%3D+${index + 1}`,
+      })
+    );
+  };
+
+  const { 1: low, 2: moderate, 3: important, 4: critical, total } = totalRisks;
+
+  const legendData = [
+    { name: `${low.title}: ${low.value}` },
+    { name: `${moderate.title}: ${moderate.value}` },
+    { name: `${important.title}: ${important.value}` },
+    { name: `${critical.title}: ${critical.value}` },
+  ];
+
+  // eslint-disable-next-line react/prop-types
+  const LegendLabel = ({ index, ...rest }) => (
+    <a key={index} onClick={() => onChartClick(null, { index })}>
+      <ChartLabel {...rest} />
+    </a>
+  );
+
+  const legend = (
+    <ChartLegend
+      height={400}
+      width={200}
+      fontSize={14}
+      data={legendData}
+      rowGutter={{ top: -5, bottom: -5 }}
+      orientation="vertical"
+      labelComponent={<LegendLabel />}
+    />
+  );
+
+  let cardBody = (
+    <ChartDonut
+      ariaDesc="Number of recommendations total-risks"
+      constrainToVisibleArea
+      data={[
+        { x: low.title, y: low.value },
+        { x: moderate.title, y: moderate.value },
+        { x: important.title, y: important.value },
+        { x: critical.title, y: critical.value },
+      ]}
+      labels={({ datum: { x, y } }) => `${x}: ${y}`}
+      legendComponent={legend}
+      legendPosition="right"
+      subTitle="Recommendations"
+      title={`${total}`}
+      padding={{
+        bottom: 20,
+        left: 20,
+        right: 140,
+        top: 20,
+      }}
+      width={350}
+      events={[
+        {
+          target: 'data',
+          eventHandlers: {
+            onClick: onChartClick,
+          },
+        },
+      ]}
+    />
+  );
+
+  if (status === STATUS.pending) {
+    cardBody = <Spinner isSVG />;
+  }
+
+  if (status === STATUS.ERROR) {
+    cardBody = <ErrorState error={error} />;
+  }
```

Comment:
do we want to keep the skeleton loading consistency on this page?
we can reuse `SkeletonLoader` component for this behavior